build_and_test.sh

```diff
@@ -91,6 +96,10 @@ inotifytools_inc="libinotifytools/src"
 inotifytools_inc2="$inotifytools_inc/inotifytools"
 inc="-I$usr_inc -I$inotifytools_inc -I$inotifytools_inc2"
 
+if [ "${ID:-linux}" = "alpine" ]; then
```

Comment:
Is it possible to fix the things pointed out as warnings instead of disabling the warning?

Reply:
fixing is hard because for that it has to get definition of the structure which is the whole problem with sys/fanotify.h mess-up that various libc implementations have done. I tried to swap the elements as well but that has same problem. Because clang does not allow VLAIS it complains about it. Usually it should work ok.

Reply:
Does this fix it? 94dfc73e7cf4a31da66b8843f0b9283ddd6b8381 

```
index f1f89132d60e2..197df344307db 100644
--- a/include/uapi/linux/fanotify.h
+++ b/include/uapi/linux/fanotify.h
@@ -162,7 +162,7 @@ struct fanotify_event_info_fid {
         * Following is an opaque struct file_handle that can be passed as
         * an argument to open_by_handle_at(2).
         */
-       unsigned char handle[0];
+       unsigned char handle[];
 };

```

maybe it doesn't, but I noticed they made this change in the kernel. I think the empty brackets is the modern way of doing it.

Reply:
Yeah that fixes it alright, this is what should be done instead of ignoring that warning... I just tested it...